polus-k-means-clustering-plugin/src/main.py

```diff
@@ -0,0 +1,244 @@
+        maximumrange(int):Ending number of sequence in range function to determine k-value.
+    Returns:
+        Optimal k-value.
+    
+    """
+    data_array = np.array(data)
+    sse = []
+    K = range(minimumrange, maximumrange)
+    for k in K:
+        kmeans = KMeans(n_clusters = k).fit(data_array)
+        centroids = kmeans.cluster_centers_
+        pred_clusters = kmeans.predict(data_array)
+        curr_sse = 0
+        # calculate square of Euclidean distance of each point from its cluster center
+        for i in range(len(data_array)):
+            curr_center = centroids[pred_clusters[i]]
+            curr_sse += (data_array[i, 0] - curr_center[0]) ** 2 + (data_array[i, 1] - curr_center[1]) ** 2
+        sse.append(curr_sse)
+    #Find the elbow point in the curve
+    points = len(sse)
+    coord = np.vstack((range(points), sse)).T
+    np.array([range(points), sse])
+    f_point = coord[0]
+    linevec = coord[-1] - coord[0]
+    linevecn = linevec / np.sqrt(np.sum(linevec**2))
+    vecf = coord - f_point
+    prod = np.sum(vecf * np.matlib.repmat(linevecn, points, 1), axis=1)
+    vecfpara = np.outer(prod, linevecn)
+    vecline= vecf - vecfpara
+    dist = np.sqrt(np.sum(vecline ** 2, axis=1))
+    k_cluster = np.argmax(dist)
+    return k_cluster
+    
+def calinski_harabasz(data, minimumrange, maximumrange):
+    """Determine k value using Calinski Harabasz Index method.
+    
+    Args:
+        data(dataframe): Input csvfile.
+        minimumrange(int): Starting number of sequence in range function to determine k-value.
+        maximumrange(int):Ending number of sequence in range function to determine k-value.
+        
+    Returns:
+        Optimal k-value.
+    
+    """
+    data_array = np.array(data)
+    if minimumrange <=1:
+        raise ValueError('Minimumrange should be greater than 1.')
+    K = range(minimumrange, maximumrange)
+    chs = []
+    for k in K:
+        kmeans = KMeans(n_clusters = k).fit(data_array)
+        labels = kmeans.labels_ 
+        ch_s = calinski_harabasz_score(data_array, labels)
+        chs.append(ch_s)
+    score = max(chs)
+    k_cluster = chs.index(score) + 1
+    return k_cluster
+
+def davies_bouldin(data, minimumrange, maximumrange):
+    """Determine k value using Davies Bouldin Index method.
+    
+    Args:
+        data(dataframe): Input csvfile.
+        minimumrange(int): Starting number of sequence in range function to determine k-value.
+        maximumrange(int):Ending number of sequence in range function to determine k-value.
+        
+    Returns:
+        Optimal k-value.
+    
+    """
+    data_array = np.array(data)
+    if minimumrange <=1:
+        raise ValueError('Minimumrange should be greater than 1.')
+    K = range(minimumrange, maximumrange)
+    dbs = []
+    for k in K:
+        kmeans = KMeans(n_clusters = k).fit(data_array)
+        preds = kmeans.fit_predict(data_array)
```

Comment:
Why are you using `fit_predict` here but above you just use `kmeans.labels_`? It seems like they perform the same task. If this is the case, you can basically just combine the davies_bouldin and calinski_harabasz methods because everything is the same except for the function used to calculate the "loss" and whether you are looking for the minimum or maximum value.